sapi/fpm/fpm/fpm_events.c

```diff
@@ -118,6 +125,9 @@ static void fpm_got_signal(struct fpm_event_s *ev, short which, void *arg) /* {{
 				}
 				/* else no access log are set */
 
+				/* We need to tie stderr with error_log in the master process after log files reload. Check #8885. */
+				fpm_stdio_init_final();
```

Comment:
The only small issue with this that it will do unnecessary extra dup2 for syslog on reload. It might be better to separate it to a new function which seems a bit cleaner as well because doing init is not exactly meant to be used here.